### DISCUSSION TITLE
Solve greatest-n-per-group with prisma.

### DISCUSSION THREAD

**Discussion by arthurfiorette** [prisma/prisma#17994] [Q&A]
Given the following `Post` model, how can I query the last _(newest / doesn't matter)_ **N** records for distinct by userId.

```prisma
model Post {
  id Int @id @unique @default(autoincrement())

  text String

  userId  Int
  user    User @relation(fields: [userId], references: [id])

  createdAt DateTime
}
```

E.g. If N is 5, how can I get the last 5 posts of each user, (If I provided 3 user Ids, should be around 15 posts)?

I'm trying to achieve a single-query solution. Is it possible with an ORM method? Or should I use `queryRaw`?

And I'm trying to avoid the following because it executes in many queries:

```js
await prisma.$transaction(
  userIds.map((userId) =>
    prisma.post.findMany({
      where: { userId },
      take: 5,
      orderBy: { createdAt: 'desc' },
    })
  )
).then(res => res.flat());
```

Reference:

https://stackoverflow.com/questions/tagged/greatest-n-per-group

**Comment by ludralph** [comment]
Hi @arthurfiorette 👋 

I think for this you can try this
```js

  const userIds = [1,2,3]

  const res = await prisma.post.findMany({
    where: { userId : { in : userIds} },
    take: 5 * userIds.length,
    orderBy: { createdAt: 'desc' },
  })
````

**Reply by arthurfiorette** [reply]
This won't work in case user 2 was the creator of the 14 last posts in chronological order. 
I need the 5 latest posts for each user.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Comment by borm** [comment]
up
